apollo-router/src/plugins/cache/invalidation.rs

```diff
@@ -184,13 +184,13 @@ async fn handle_request(
                         .map(|k| RedisKey(k.to_string()))
                         .collect::<Vec<_>>();
                     if !keys.is_empty() {
-                        count += keys.len() as u64;
-                        storage.delete(keys).await;
+                        let len = keys.len() as u64;
+                        count += storage.delete(keys).await.unwrap_or(0) as u64;
 
                         u64_counter!(
                             "apollo.router.operations.entity.invalidation.entry",
                             "Entity cache counter for invalidated entries",
-                            1u64,
+                            len,
```

Comment:
What is this counter measuring? Scanned or deleted?

I think it should be "deleted", so:

```
let len = storage.delete(keys).await.unwrap_or(0) as u64;
count += len
```

Reply:
what do you mean? Updating the description?

Reply:
I mean `keys.len()` isn't the same as `storage.delete(keys)...`. That's all. If the motivation is to have a counter that displays how many entries were invalidated, should it be "what we deleted" or "what we scanned". Honestly, I'm not sure, maybe it is better to have it as scanned, since something else might have deleted the records, but ...

Reply:
it's more useful to have the number of actually deleted entries yes

Reply:
fixed in [37fb0f0](https://github.com/apollographql/router/pull/5890/commits/37fb0f0ab4ac9bd5abcf5ae994a3c9c441d4fa3c)